File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.java
import java.util.*;
class Solution {
    public String[] solution(String[] record) {
        String[] answer = {};
        List<String> answerList = new ArrayList<>();
        HashMap<String,String> map = new HashMap<>();
        
        for(int i=0;i<record.length;i++){
            String[] tmp = record[i].split(" ");
            if(tmp[0].equals("Enter")){
                map.put(tmp[1],tmp[2]);
            }else if(tmp[0].equals("Change")){
                map.put(tmp[1],tmp[2]);
            }
        }
        for(int i=0; i<record.length;i++){
            String[] arr = record[i].split(" ");
            if(arr[0].equals("Enter")){
                answerList.add(map.get(arr[1])+"님이 들어왔습니다.");
            }else if(arr[0].equals("Leave")){
                answerList.add(map.get(arr[1])+"님이 나갔습니다.");
            }
        }
        answer = answerList.stream().toArray(String[]::new);
        
    return answer;
    }
}

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    dict = {}
    for i in range(len(record)):
        now = record[i].split(' ')
        
        if now[0] == 'Leave':
            continue
        dict[now[1]] = now[2]

    for r in record:
        now = r.split(' ')
        s = dict[now[1]]
        if now[0] == 'Enter':
            s += '님이 들어왔습니다.'
            answer.append(s)
            
        elif now[0] == 'Leave':
            s += '님이 나갔습니다.'
            answer.append(s)
        
        
    return answer